apps/server/src/modules/user/service/user.service.ts

```diff
@@ -9,31 +9,42 @@ import { ConfigService } from '@nestjs/config';
 import { DataDeletionDomainOperationLoggable } from '@shared/common/loggable';
 import { Page, RoleReference, UserDO } from '@shared/domain/domainobject';
 import { LanguageType, User } from '@shared/domain/entity';
-import { DomainOperation, IFindOptions } from '@shared/domain/interface';
+import { DeletionService, DomainDeletionReport, DomainOperationReport, IFindOptions } from '@shared/domain/interface';
 import { DomainName, EntityId, OperationType, StatusModel } from '@shared/domain/types';
 import { UserRepo } from '@shared/repo';
 import { UserDORepo } from '@shared/repo/user/user-do.repo';
 import { Logger } from '@src/core/logger';
-import { DomainOperationBuilder } from '@shared/domain/builder';
+import { DomainDeletionReportBuilder, DomainOperationReportBuilder } from '@shared/domain/builder';
 import { DeletionErrorLoggableException } from '@shared/common/loggable-exception';
+import { EventBus, EventsHandler, IEventHandler } from '@nestjs/cqrs';
+import { DataDeletedEvent, UserDeletedEvent } from '@modules/deletion';
+import { RegistrationPinService } from '@modules/registration-pin';
 import { UserConfig } from '../interfaces';
 import { UserMapper } from '../mapper/user.mapper';
 import { UserDto } from '../uc/dto/user.dto';
 import { UserQuery } from './user-query.type';
 
 @Injectable()
-export class UserService {
+@EventsHandler(UserDeletedEvent)
+export class UserService implements DeletionService, IEventHandler<UserDeletedEvent> {
 	constructor(
 		private readonly userRepo: UserRepo,
 		private readonly userDORepo: UserDORepo,
 		private readonly configService: ConfigService<UserConfig, true>,
 		private readonly roleService: RoleService,
 		private readonly accountService: AccountService,
-		private readonly logger: Logger
+		private readonly registrationPinService: RegistrationPinService,
```

Comment:
@Metauriel unfortunately it cannot, at least not through the user data deletion event. _Registration pins_ domain layer can be considered a child of the _users_ domain layer, but it's not mapped directly through the _userId_ (it would be possible then), but through the _email address_. And email address is a part of the _user_ entity. So the _user_ entity deletion should fail in case of a failed _registrationpins_ deletion - otherwise we won't be able to find the email addresses when we'll want to retry the _registrationpins_ deletion.

The only way I can see now to handle _registrationspins_ deletion separately would be to create another type of a deletion request and an event (_emailDataDeletion_ or something like that, that would contain the email for which we want to delete all the existing data). Without it we should keep the _registrationpins_ deletion process within the _users_ entities deletion process for now as it's the only safe way to make sure we will be able to get rid of the _registrationpins_ in case retrying this operation is needed.